torchsparse/nn/functional/conv.py

```diff
@@ -107,34 +131,42 @@ def conv3d(inputs,
         output_tensor.kernel_maps = inputs.kernel_maps
         output_tensor.check()
     elif not transpose:
-        kernel_map = inputs.kernel_maps.get(
-            'k%s_os%d_s%d_d%d' % (kernel_size, cur_stride, stride, dilation),
-            None)
+        kernel_map_key = KernelMapConfig(kernel_size, cur_stride, stride,
+                                         dilation)
+        kernel_map = inputs.kernel_maps.get(kernel_map_key, None)
+
+        do_downsample = any(x > 1 for x in stride)
 
-        if stride > 1:
+        if do_downsample:
             # do downsample
             kRegion = KernelRegion(kernel_size=kernel_size,
                                    tensor_stride=cur_stride)
             kOffset = kRegion.get_kernel_offset().to(features.device)
-            new_coords = spdownsample(coords, stride * cur_stride)
+            new_coords = spdownsample(
+                coords,
+                torch.IntTensor(stride).to(features.device).unsqueeze(0),
+                kernel_size,
+                torch.IntTensor(cur_stride).to(features.device).unsqueeze(0))
```

Comment:
Would it be possible to move these type castings inside `spdownsample`?

Reply:
Yes, I just move them into `spdownsample`.